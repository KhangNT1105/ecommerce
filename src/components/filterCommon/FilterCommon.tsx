import React from 'react';

import Styles from './FilterCommon.module.scss';

import FilterBlock from '../../components/filterBlock/FilterBlock';
import FilterTag from 'components/filterTag/FilterTag';

interface SuggestionWithTitle {
  title: string;
  value: string;
}
interface FilterValue {
  tagName: string;
  values: string[];
  suggestionList?: (string | SuggestionWithTitle)[];
  hasInput?: boolean;
  apiLookupField?: string;
  fieldName?: string;
}

interface FilterCommonProps {
  filterList: string[];
  filterValueList: FilterValue[];
  onFieldChoose?: (params: any) => void;
  addValue?: (field: string, value: string) => void;
  removeValue?: (field: string, value: string) => void;
  removeField?: (field: string) => void;
  className?: string;
}

const FilterCommon: React.FC<FilterCommonProps> = ({
  filterList,
  filterValueList,
  onFieldChoose,
  addValue,
  removeValue,
  removeField,
  className
}) => {
  return (
    <div className={`${Styles['filter-common']} ${className}`}>
      <FilterBlock filterList={filterList} onItemChoose={onFieldChoose} />
      {filterValueList.map(({ apiLookupField, fieldName, tagName, values, suggestionList, hasInput }, index) => {
        return (
          <FilterTag
            className={Styles['filter-tag']}
            endpoint={apiLookupField}
            fieldName={fieldName}
            hasInput={hasInput}
            key={index}
            tagName={tagName}
            suggestionList={suggestionList}
            values={values}
            addValue={addValue}
            removeValue={removeValue}
            removeField={removeField}
          />
        );
      })}
    </div>
  );
};

export default FilterCommon;
