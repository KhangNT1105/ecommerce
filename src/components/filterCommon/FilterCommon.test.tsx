import React from 'react';
import renderer from 'react-test-renderer';
import FilterCommon from './FilterCommon';

jest.mock('../filterBlock/FilterBlock', () => 'FilterBlock').mock('../filterTag/FilterTag', () => 'FilterTag');

describe('component FilterCommon', () => {
  test('renders snapshot correctly', () => {
    const tree = renderer
      .create(
        <FilterCommon
          filterList={['id', 'name', 'this is a very long item']}
          filterValueList={[
            {
              tagName: 'test',
              values: ['test'],
              suggestionList: ['test']
            }
          ]}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
