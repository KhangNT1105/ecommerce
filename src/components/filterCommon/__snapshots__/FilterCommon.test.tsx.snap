// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component FilterCommon renders snapshot correctly 1`] = `
<div
  className="filter-common undefined"
>
  <FilterBlock
    filterList={
      Array [
        "id",
        "name",
        "this is a very long item",
      ]
    }
  />
  <FilterTag
    className="filter-tag"
    suggestionList={
      Array [
        "test",
      ]
    }
    tagName="test"
    values={
      Array [
        "test",
      ]
    }
  />
</div>
`;
